modules/test/playwright/tests/frontend-data-set-admin-web/tests/data-set-admin/dataSets.spec.ts

```diff
@@ -150,6 +163,131 @@ async function assertTableRowsCount(page, rowsCount) {
 	});
 }
 
+async function setupUserRoleAndLoginAsUser({
+	apiHelpers,
+	dataSetResourcePermissions,
+	page,
+	roleDefinePermissionsPage,
+	rolePage,
+	rolesPage,
+}: {
+	apiHelpers: DataApiHelpers;
+	dataSetResourcePermissions?: {
+		actions: string[];
+		name: string;
+	}[];
+	page: Page;
+	roleDefinePermissionsPage: RoleDefinePermissionsPage;
+	rolePage: RolePage;
+	rolesPage: RolesPage;
+}) {
+	const roleName = `ds_user_${getRandomString()}`;
+
+	let dataSetUserRole;
+	let userAccount: TUserAccount;
+
+	await test.step('Create Data Set user role', async () => {
+		const companyId = await page.evaluate(() => {
+			return Liferay.ThemeDisplay.getCompanyId();
+		});
+
+		dataSetUserRole = await apiHelpers.headlessAdminUser.postRole({
+			name: roleName,
+			rolePermissions: [
+				{
+					actionIds: ['VIEW_CONTROL_PANEL'],
+					primaryKey: companyId,
+					resourceName: '90',
```

Comment:
why 90?

Reply:
Apparently it's the key for Portal, not sure how that number was chosen though:

https://github.com/liferay/liferay-portal/blob/master/portal-kernel/src/com/liferay/portal/kernel/util/PortletKeys.java#L132